extensions/2.0/Vendor/EXT_feature_metadata/examples/texture-metadata.gltf

```diff
@@ -0,0 +1,257 @@
+      "byteOffset": 180,
+      "byteLength": 72
+    },
+    {
+      "buffer": 0,
+      "byteOffset": 252,
+      "byteLength": 48
+    },
+    {
+      "buffer": 0,
+      "byteOffset": 300,
+      "byteLength": 48
+    },
+    {
+      "buffer": 0,
+      "byteOffset": 348,
+      "byteLength": 24
+    }
+  ],
+  "buffers": [
+    {
+      "byteLength": 372,
+      "uri": "external.bin"
+    }
+  ],
+  "samplers": [
+    {
+      "magFilter": 9728,
+      "minFilter": 9728
+    }
+  ],
+  "images": [
+    {
+      "uri": "color.jpg"
+    },
+    {
+      "uri": "lulc.png"
+    },
+    {
+      "uri": "ndvi.jpg"
+    }
+  ],
+  "textures": [
+    {
+      "source": 0
+    },
+    {
+      "source": 1,
+      "sampler": 0,
+      "extensions": {
+        "EXT_feature_metadata": {
+          "featureMappings": [
+            {
+              "featureCollection": 0,
+              "featureIds": {
+                "channel": "r"
+              }
+            }
+          ]
+        }
+      }
+    },
+    {
+      "source": 2,
+      "extensions": {
+        "EXT_feature_metadata": {
+          "features": [
+            {
+              "featureClass": "NDVI",
+              "properties": {
+                "NDVI": {
+                  "channels": "r"
+                }
+              }
+            }
+          ]
+        }
+      }
+    }
+  ],
+  "materials": [
+    {
+      "pbrMetallicRoughness": {
+        "baseColorTexture": {
+          "index": 0,
+          "texCoord": 0
+        }
+      },
+      "extensions": {
+        "EXT_feature_metadata": {
+          "textures": [
+            {
+              "index": 1,
+              "texCoord": 1
+            },
+            {
+              "index": 2,
+              "texCoord": 1
+            }
+          ]
+        }
+      }
+    }
+  ],
+  "meshes": [
+    {
+      "primitives": [
+        {
+          "attributes": {
+            "POSITION": 0,
+            "TEXCOORD_0": 1,
+            "TEXCOORD_1": 2
+          },
+          "indices": 3,
+          "material": 0,
+          "mode": 4
+        }
+      ]
+    }
+  ],
+  "extensions": {
+    "EXT_feature_metadata": {
+      "extras": {
+        "draftVersion": "0.0.0"
+      },
+      "featureClasses": {
+        "NDVI": {
+          "name": "Normalized difference vegetation index",
+          "semantic": "NDVI",
+          "properties": {
+            "NDVI": {
+              "semantic": "NDVI",
+              "valueType": "UINT8",
+              "normalized": true
+            }
+          }
+        },
+        "LULC": {
+          "name": "Landcover/Landuse",
+          "semantic": "LULC",
+          "properties": {
+            "Name": {
+              "name": "Name",
+              "semantic": "NAME",
+              "valueType": "STRING"
+            },
+            "Color": {
+              "name": "Color",
+              "semantic": "COLOR",
+              "valueType": "UINT8",
+              "valueCount": 3
```

Comment:
I found the naming conventions for `valueType`, `valueCount`, and `elementType` somewhat confusing.

I talked about this with @lilleyse, he suggested I try to think of a better naming convention. I'll update this when I have any ideas.

Reply:
Here's my proposal for naming convention. Though note that I'd need to learn how to phrase this properly in JSON Schema.

Here's a pseudo-grammar for this:

```
valueType := ARRAY <arrayProperties> | VARIABLE_SIZE_ARRAY <varArrayProperties> | STRING | <primitiveType>

# additional keys you'd need alongside valueType
arrayProperties := {
    componentType: <primitiveType>, 
    componentsPerElement: integer
}

varArrayProperties := {
    componentType: <primitiveType>,
   # not enforcible here, but you also need an offsetBufferView
}

# just a helper enum
primitiveType := `BOOLEAN | *INT* | FLOAT*`
```

Examples:

```
// single uint16 property
{
  "name": "
  "valueType": "UINT16"
}

// vec3 property (
{
  "name": 
  "valueType": "ARRAY",
  "componentType": "FLOAT32
}
```

**TODO** Finish this comment after calls are done

Reply:
While writing the above comment, I'm finding I'm not 100% sure how would you use variable length arrays besides binary and string.

@lilleyse could you provide any further examples on this

Reply:
One example: say your feature class is "Tree" and one of the properties is "Bird Species", a list of integer codes of bird species that inhabit that tree.

We also want to be able to support variable-sized arrays if we get around to the idea of packing the tileset.json tree into binary where each tile stores indices to its children.

Reply:
**glTF / OpenGL terminology** (slightly different use case)

Attribute
Element
Component

**Apache Arrow**

Field
Value
Slot

**Current terminology**

Property
Element
Value

**Proposed terminology**

Property
Value
Component

Reply:
We settled on:

* Property -- because attribute could be confused with vertex attribute
* Element -- even when the element is a string/array, it's still an element of the broadest array (the feature collection table's "column")
* Component -- because this implies a part of a larger value (like a component of a vec3)